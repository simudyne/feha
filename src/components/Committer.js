import * as React from 'react'
import {Image} from 'semantic-ui-react'

const style = {
  color: '#586069',
  fontSize: '.8rem'
}

export const Committer = ({author, committer, commit}) => {

  if (author)
    return <div className="author" style={style}>
      <Image src={author.avatar_url} avatar />
      <a href={author.html_url} target="_blank">{author.login}</a>
    </div>

  if (committer)
    return <div className="committer" style={style}>
      <Image src={committer.avatar_url} avatar />
      <a href={committer.html_url} target="_blank">{committer.login}</a>
    </div>

  const authored = <a href={`mailto:${commit.author.email}`}>{commit.author.name}</a>
  const committed = <a href={`mailto:${commit.committer.email}`}>{commit.committer.name}</a>
  const avatarUrl = 'https://cdn1.iconfinder.com/data/icons/user-pictures/100/unknown-512.png'
  return <div className="commit" style={style}>
    <Image src={avatarUrl} avatar />
    {authored} authored and {committed} committed
  </div>
}
