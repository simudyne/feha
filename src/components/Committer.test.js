import * as React from 'react'
import { shallow } from 'enzyme'
import { Committer } from './Committer'
import renderer from 'react-test-renderer'

const author = {
  login: 'JohnDoe',
  avatar_url: 'http://avatar.com/JohnDoe',
  html_url: 'http://github.com/JohnDoe'
}

const committer = {
  login: 'JaneDoe',
  avatar_url: 'http://avatar.com/JaneDoe',
  html_url: 'http://github.com/JaneDoe'
}

const commit = {
  author: { name: author.login, email: author.html_url },
  committer: { name: committer.login, email: committer.html_url }
}

describe('Committer component', () => {

  it('Should display author', () => {
    const props = { author }
    const comp = <Committer {...props} />
    const wrapper = shallow(comp)
    const tree = renderer.create(comp).toJSON()
    expect(wrapper.find('.author').exists()).toBe(true)
    expect(wrapper.find('.committer').exists()).toBe(false)
    expect(wrapper.find('.commit').exists()).toBe(false)
    expect(tree).toMatchSnapshot()
  })

  it('Should display committer', () => {
    const props = { committer }
    const comp = <Committer {...props} />
    const wrapper = shallow(comp)
    const tree = renderer.create(comp).toJSON()
    expect(wrapper.find('.author').exists()).toBe(false)
    expect(wrapper.find('.committer').exists()).toBe(true)
    expect(wrapper.find('.commit').exists()).toBe(false)
    expect(tree).toMatchSnapshot()
  })

  it('Should display commits author and committer', () => {
    const props = { commit }
    const comp = <Committer {...props} />
    const wrapper = shallow(comp)
    const tree = renderer.create(comp).toJSON()
    expect(wrapper.find('.author').exists()).toBe(false)
    expect(wrapper.find('.committer').exists()).toBe(false)
    expect(wrapper.find('.commit').exists()).toBe(true)
    expect(tree).toMatchSnapshot()
  })
})