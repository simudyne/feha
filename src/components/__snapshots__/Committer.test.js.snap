// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Committer component Should display author 1`] = `
<div
  className="author"
  style={
    Object {
      "color": "#586069",
      "fontSize": ".8rem",
    }
  }
>
  <img
    className="ui avatar image"
    src="http://avatar.com/JohnDoe"
  />
  <a
    href="http://github.com/JohnDoe"
    target="_blank"
  >
    JohnDoe
  </a>
</div>
`;

exports[`Committer component Should display commits author and committer 1`] = `
<div
  className="commit"
  style={
    Object {
      "color": "#586069",
      "fontSize": ".8rem",
    }
  }
>
  <img
    className="ui avatar image"
    src="https://cdn1.iconfinder.com/data/icons/user-pictures/100/unknown-512.png"
  />
  <a
    href="mailto:http://github.com/JohnDoe"
  >
    JohnDoe
  </a>
   authored and 
  <a
    href="mailto:http://github.com/JaneDoe"
  >
    JaneDoe
  </a>
   committed
</div>
`;

exports[`Committer component Should display committer 1`] = `
<div
  className="committer"
  style={
    Object {
      "color": "#586069",
      "fontSize": ".8rem",
    }
  }
>
  <img
    className="ui avatar image"
    src="http://avatar.com/JaneDoe"
  />
  <a
    href="http://github.com/JaneDoe"
    target="_blank"
  >
    JaneDoe
  </a>
</div>
`;
